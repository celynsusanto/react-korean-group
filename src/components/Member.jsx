import React from 'react'
import ApplicationContext from '../context'

const generatePadding = index => {
  const getIndex = generateIndex(index)
  const dictionary = {
    0: 'pr-2',
    1: 'px-1',
    2: 'px-1',
    3: 'pl-2'
  }

  return `w-1/4 ${dictionary[getIndex]} pb-4`
}

const generateIndex = index => {
  if (index <= 3) {
    return index
  }
  return generateIndex(index - 4)
}

export default function Member({member, index}) {
  return (
    <ApplicationContext.Consumer>
      {theme => (
        <div className={generatePadding(index)}>
          <div className={`${theme.card} rounded shadow flex flex-col justify-center`}>
            <img className="cover w-full" src={member.image} alt="apink" />
            <span className={`font-medium text-xl text-center p-2 ${theme.text}`}>{member.name}</span>
            <span className={`font-medium text-center p-2 ${theme.text}`}>{member.dateOfBirth}</span>
          </div>
        </div>
      )}
    </ApplicationContext.Consumer>
  )
}
