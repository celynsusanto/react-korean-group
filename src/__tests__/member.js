import {render} from 'react-testing-library'
import React from 'react'
import Member from '../components/Member.jsx'

test('it will render member component', () => {
  const member = {
    image: '/',
    name: 'Kim Nam Joo',
    dateOfBirth: '15-04-1995'
  }
  const {getByTestId, debug} = render(<Member index={0} member={member} />)
  // Code ini akan nge print dom kita saat fungsi ini dipanggil
  // debug()

  // bisa juga memanggil debug pada element yang kita inginkan
  // debug(getByTestId('member-name'))

  expect(getByTestId('member-name')).toHaveTextContent(member.name)
  expect(getByTestId('member-image')).toHaveStyle(`background-image : url(${member.image})`)
  expect(getByTestId('member-date-of-birth')).toHaveTextContent(member.dateOfBirth)

  expect(getByTestId('member-container')).toHaveClass('w-1/4 pr-2 pb-4')
})

test('it will match the snapshot', () => {
  const member = {
    image: '/',
    name: 'Kim Nam Joo',
    dateOfBirth: '15-04-1995'
  }

  const {container} = render(<Member index={0} member={member} />)

  expect(container).toMatchSnapshot()
})
