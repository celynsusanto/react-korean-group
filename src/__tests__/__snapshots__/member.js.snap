// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it will match the snapshot 1`] = `
<div>
  <div
    class="w-1/4 pr-2 pb-4"
    data-testid="member-container"
    id="asd"
  >
    <div
      class="bg-white rounded shadow flex flex-col justify-center"
    >
      <div
        class="bg-cover w-full h-big block"
        data-testid="member-image"
        style="background-position: center; background-image: url(/);"
      />
      <span
        class="font-medium text-xl text-center p-2"
        data-testid="member-name"
      >
        Kim Nam Joo
      </span>
      <span
        class="font-medium text-subtle-grey text-center p-2"
        data-testid="member-date-of-birth"
      >
        15-04-1995
      </span>
    </div>
  </div>
</div>
`;
